src/discovery/discoveryLogic.ts

```diff
@@ -1,90 +1,247 @@
 
-type TypeIndex = {  label: string, index: NamedNode, agent: NamedNode } ;
+const {  getContainerMembers, authn } = solidLogicSingleton
+const { currentUser } = authn
+
+type TypeIndexScope = { label: string, index: NamedNode, agent: NamedNode } ;
+type ScopedApp = { instance: NamedNode, scope: TypeIndexScope }
 
 const ns ={
+  rdf:   Namespace('http://www.w3.org/1999/02/22-rdf-syntax-ns#'),
   solid: Namespace('http://www.w3.org/ns/solid/terms#'),
   space: Namespace('http://www.w3.org/ns/pim/space#')
 }
 
-export async function loadProfile(store: LiveStore, user) {
+/** Create a resource if it really does not exist
+ *  Be absolutely sure something does not exist before creating a new empty file
+ * as otherwise existing could  be deleted.
+ * @param doc {NamedNode} - The resource
+ */
+export async function loadOrCreateIfNotExists (store: LiveStore, doc: NamedNode) {
+    let response
+    try {
+      response = await store.fetcher.load(doc)
+    } catch (err) {
+        if (err.response.status === 404) {
+          debug.log('createIfNotExists doc does NOT exist, will create: ' + doc)
+          try {
+            store.fetcher.webOperation('PUT', doc.uri, {data: '', contentType: 'text/turtle'})
+          } catch (err) {
+            const msg = 'createIfNotExists doc FAILED: ' + doc + ': ' + err
+            debug.log(msg)
+            throw new Error(msg)
+          }
+          delete store.fetcher.requested[doc.uri] // delete cached 404 error
+          debug.log('createIfNotExists doc created ok ' + doc)
+        } else {
+          const msg =  'createIfNotExists doc load error NOT 404:  ' + doc + ': ' + err
+          debug.log(msg)
+          throw new Error(msg) // @@ add nested errors
+        }
+    }
+    debug.log('createIfNotExists doc exists, all good ' + doc)
+    return response
+}
+
+export function HTTPStatus (doc: NamedNode): Number {
+  return 200 // @@ TBD
+}
+export function makePreferencesFileURI (store:LiveStore, me:NamedNode) {
+  const stripped = me.uri.replace('/profile/', '/').replace('/public/', '/')
+  // const stripped = me.uri.replace(\/[p|P]rofile/\g, '/').replace(\/[p|P]ublic/\g, '/')
+  const folderURI = stripped.split('/').slice(0,-1).join('/') + '/Settings/'
+  const fileURI = folderURI + 'Preferences.ttl'
+  const folder = sym(folderURI)
+
+  const folderStat = HTTPStatus(folder)
+  if (folderStat === 200) {
+
+  } else if (folderStat === 404) {
+
+  } else {
+    console.log('Abort Preferences creation: Settings folder HTTP status: ' + folderStat)
+    return null
+  }
+  // @@ To be completed TBC
+
+}
+/* Follow link from this doc to another thing, or else make a new link
+**
+**  null return means new one created
+*/
+export async function followOrCreateLink(store: LiveStore, subject: NamedNode, predicate: NamedNode,
+     object: NamedNode, doc:NamedNode):Promise<NamedNode | null> {
+  const result = store.any(subject, predicate, null, doc)
+  if (result) return result as NamedNode
```

Comment:
Shouldn't we be interested to check if result === object? 
What if they are not equal?